### DISCUSSION TITLE
How to make a ListView/GridView work with VS Themes

### DISCUSSION THREAD

**Discussion by awschristou** [VsixCommunity/Community.VisualStudio.Toolkit#321] [Q&A]
I would like to use ListViews (containing GridViews) and have them leverage the VS Themes. I've added them to the Test extension's Theme Window [in my fork](https://github.com/awschristou/Community.VisualStudio.Toolkit/commit/7c715cd1d966f2d383aa580e4ee1e357d24a118f).

I notice two issues:
- when using the VS Theme, the column headings are not visible. How can I show the headings?
- when using the VS Theme, the list items are not using the VS Theme (the biggest tell is that the selected row has a gradient). Would it be possible for the Toolkit to handle theme support for these?

Here is what they look like in a combination of Blue/Dark, with the theme on/off:
![image](https://user-images.githubusercontent.com/39839589/163069134-8de0787b-129e-49df-afd2-238d875139a5.png)

**Comment by awschristou** [comment]
@reduckted any chance you'd have some insights on this one?

**Reply by reduckted** [reply]
Hmm, using [Snoop](https://github.com/snoopwpf/snoopwpf) you can see the style and template are changing (highlighted in yellow) when enabling theming.
![listview](https://user-images.githubusercontent.com/10321525/164453768-3cf2c26f-74b3-4499-8357-7ffd58aaf399.gif)
 
At a guess, I would say that Visual Studio is applying a default style to `ListViews` that causes the header to either be hidden or removed completely via the `ControlTemplate`. I'm yet to actually work out where that style is defined. I'm pretty sure it will be in one of the assemblies under `<InstallDir>\Common7\IDE`, but that doesn't narrow it down too much 😆

**Comment by jarnovanderlinden** [comment]
I have the same problem using VS2022. I have seen the header row be zero height, but also in other cases be the correct height but with no text or column dividers showing. And like yours the selection highlighting is an unusable white bar on dark theme.

**Reply by Max-Cadenas** [reply]
Did you find a solution or maybe a quick workaround? I'm running into the header issue.

**Reply by awschristou** [reply]
I ended up defining my own style for TargetType `GridViewColumnHeader`, `ListView`, and `ListViewItem`. I don't recall if you would only need to define one for GridViewColumnHeader as a quick workaround.

**Comment by reduckted** [comment]
Instead of using a `ListView`, perhaps using Visual Studio's table control would be a better solution?

Here's a demo that I've managed to hack together that shows the current projects in a table.

![image](https://user-images.githubusercontent.com/10321525/164956910-1a9e3bda-7e10-46c9-a956-a5828d85fa6d.png)

A word of warning though. It seems very complicated, and as far as I can tell, there is _zero_ documentation on how to do it. I managed to hack that together by deciphering the [EditorConfigSettings](https://github.com/dotnet/roslyn/tree/5bce330dc6294f059e6cf69a2ee77b02e141c804/src/VisualStudio/Core/Def/EditorConfigSettings) UI. 

I think @madskristensen logged #82 quite a while ago to create a wrapper around all of the bits and pieces that go into creating a table control.
